tests/components/nice_go/test_event.py

```diff
@@ -10,13 +13,25 @@
 
 
 async def test_barrier_obstructed(
-    hass: HomeAssistant, mock_nice_go: AsyncMock, mock_config_entry: MockConfigEntry
+    hass: HomeAssistant,
+    mock_nice_go: AsyncMock,
+    mock_config_entry: MockConfigEntry,
+    freezer: FrozenDateTimeFactory,
 ) -> None:
     """Test barrier obstructed."""
     mock_nice_go.event = MagicMock()
-    await setup_integration(hass, mock_config_entry, ["event"])
+    await setup_integration(hass, mock_config_entry, [Platform.EVENT])
 
     await mock_nice_go.event.call_args_list[2][0][0]({"deviceId": "1"})
     await hass.async_block_till_done()
 
-    assert hass.states.get("event.nice_go_1_barrier_obstructed") != "unknown"
+    assert (
+        hass.states.get("event.test_garage_1_barrier_obstructed").state
+        == freezer.time_to_freeze.isoformat(timespec="milliseconds") + "+00:00"
```

Comment:
Just set the time to a specific point, checkout withings/test_sensor.py, and then the first test